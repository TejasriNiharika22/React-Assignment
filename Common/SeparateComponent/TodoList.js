import React from "react";
import Edit from './Edit';

class TodoList extends React.Component {
  renderItems() {
    return this.props.data.map((item, index) => <Edit key={index} {...item} {...this.props} />);
  }
  render() {
    return (
      <div className="list">
        {this.renderItems()}
      </div>
    );
  }
}
export default TodoList;