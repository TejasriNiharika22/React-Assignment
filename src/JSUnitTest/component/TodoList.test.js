import { shallow } from "enzyme";
import TodoList from '../../components/TodoList';

describe('<TodoList />', () => {
    it('should render properly', () => {
        const wrapper = shallow(<TodoList data={[]} />);
        expect(wrapper).toMatchSnapshot(); //Snapshots allows to test if the component renders correctly
    });
    it('should test <TodoList> component with list of todos', () => {
        const wrapper = shallow(
            <TodoList data={['Doing Exercise', 'Wake up early']} />
        );
        expect(wrapper.find('.list').exists()).toBe(true);
    });
});