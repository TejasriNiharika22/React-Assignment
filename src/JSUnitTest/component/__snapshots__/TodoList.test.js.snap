// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoList /> should render properly 1`] = `ShallowWrapper {}`;

exports[`<TodoList /> should test <TodoList> component with list of todos 1`] = `ShallowWrapper {}`;
